src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -103,13 +107,30 @@ public ToolTipProvider getToolTipProvider() {
      * Returns the associated build of this action.
      *
      * @return the associated build of this action
+     * @deprecated use {@link #getBuild()} instead
      */
+    @Deprecated
+    @CheckForNull
```

Comment:
I still think this should be 

```
@WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
public final Run<?,?> getOwner() {
  return owner;
}

@WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getAbstractBuild")
@Deprecated
public final Run<?,?> getBuild() {
  return owner;
}

@Restricted(NoExternalUse.class) // I think this could be private but if not then restrict it
@Deprecated
public final AbstractBuild<?, ?> getAbstractBuild() {
  return owner instanceof AbstractBuild ? (AbstractBuild<?, ?>) owner : null;
}
```

I don't like the chain of migrations from build -> owner -> run... it makes it harder for people to conceptualize.

You will likely also need to update the version of the bridge-method-injector plugin in the pom from the inherited version to 1.14 to get support for adapterMethod